### ISSUE TITLE
Shopping cart implementation

### ISSUE THREAD

**Issue by Thomaxo** [impresspages/ImpressPages#697]
I was wondering on how to implement a shopping cart, either by a plug-in or maybe some pointers on how to modify the code. Also i was wondering if you could maybe reuse a product on another page.

Thanks a lot in advance,
Thomas

**Comment by maskas** [comment]
Simple Product plugin would be a good start for a shopping cart http://market.impresspages.org/plugins/SimpleProduct

**Comment by Thomaxo** [comment]
Thanks for your reply, Maskas.
I've already found out about SimpleProduct and the payment modules. They're pretty awesome.
I say "pretty" because i'm missing some basic functionality like a shopping cart.
Currently SimpleProduct only allows 1 product purchase at a time. There is no ability to add another into a shopping cart, which i find kind of odd. I'm in love with impress pages(i barely know php and i can kind of understand how it works.. simplicity at its best) and the back-end management is awesome.

So basically i was looking into adding a shopping cart functionality, instead of immediate payment.
I think i'll manage to write some adding or subtracting code in php, but mostly worried about how this will affect the payment modules.
For this i'd love some pointers, how will this affect the payment modules, do i better look elsewhere, etc.

**Comment by Thomaxo** [comment]
Never mind! :) I found another php shopping cart and i believe i'll manage to integrate that in impress pages; it is a php framework after all.

The only question is: can i integrate that with the existing payment modules? Multiple products and such..?

Thanks,
Thomas

**Comment by maskas** [comment]
The payments in ImpressPages are done in a way that it doesn't matter how many products there are (one or many). You just say, that you need some amount of money to be paid and the ID of the payment and the rest is done by the payment method https://www.impresspages.org/docs/payment-methods

So if you are implementing a shopping cart, then you supply the total price of the whole shopping cart to the payment method and that's it.

The plan to integrate a shopping cart is a good idea. But I don't think you will be able to use it's shopping cart and ImpressPages payment. If you use a third party shopping cart, probably all product related pages and the payment will be handled by the shopping cart.

**Comment by Thomaxo** [comment]
In your opinion, is it possible to integrate php tags from 2 different CMS'es?
Like, the in the "home.php", could i also add a few php tags from OpenCart?
Or would i be better off templating it all with pure html?
OpenCart seems to, after looking through the files, be pretty modular.

Thanks.

**Comment by bekoli** [comment]
This can _probably_ be done. I say _probably_ because it depends a lot on the integration features of OpenCart (that I do not know).

I'm interested in a cart system as well. I can't dig into this now however. Probably not within the next 6 months ;)

I can however strongly advise you to create the cart as Plugin instead of putting it's logic into your theme. This is a trap that will trigger on some update one day and leave you with a broken shop.

From history:

I managed to add Phorum to IP3 two years ago. That's a full featured forum software. Basically I had to include some `common.php` (or else) that took care of the class loading (or including) and I was set. I remember some url override in Phorum as well, that I realized as plugin (this thing has hooks *everywhere*), to not break future updates.

The tricky part was to tell IP to forward my /forum/ route and to integrate the search function. You may not need such though.

Sadly I had to drop this project due to time and when I started with IP4 again things changed quite a bit (also Phorum appears to be dead in the water).
